detector/aws_db_instance_previous_type.go

```diff
@@ -9,22 +9,21 @@ import (
 
 type AwsDBInstancePreviousTypeDetector struct {
 	*Detector
-	IssueType             string
-	TargetType            string
-	Target                string
-	DeepCheck             bool
 	previousInstanceTypes map[string]bool
 }
 
 func (d *Detector) CreateAwsDBInstancePreviousTypeDetector() *AwsDBInstancePreviousTypeDetector {
-	return &AwsDBInstancePreviousTypeDetector{
+	nd := &AwsDBInstancePreviousTypeDetector{
 		Detector:              d,
-		IssueType:             issue.WARNING,
-		TargetType:            "resource",
-		Target:                "aws_db_instance",
-		DeepCheck:             false,
 		previousInstanceTypes: map[string]bool{},
 	}
+	nd.Name = "aws_db_instance_previous_type"
+	nd.IssueType = issue.WARNING
+	nd.TargetType = "resource"
```

Comment:
__[gometalinter]__
* [goconst] 45 other occurrence(s) of "resource" found in: aws_instance_invalid_type.go:22:18 aws_alb_invalid_subnet.go:23:18 aws_instance_not_specified_iam_profile.go:16:18 aws_route_specified_multiple_targets.go:16:18 aws_elasticache_cluster_duplicate_id.go:22:18 aws_alb_invaid_security_group.go:23:18 aws_route_invalid_nat_gateway.go:22:18 aws_db_instance_invalid_type.go:22:18 aws_route_invalid_network_interface.go:22:18 test_helper.go:71:7 aws_db_instance_readable_password.go:16:18 aws_db_instance_invalid_option_group.go:22:18 aws_elb_duplicate_name.go:22:18 aws_db_instance_invalid_vpc_security_group.go:23:18 aws_cloudwatch_metric_alarm_invalid_unit.go:22:18 aws_route_not_specified_target.go:16:18 aws_instance_invalid_subnet.go:22:18 aws_db_instance_default_parameter_group.go:19:18 aws_elasticache_cluster_invalid_parameter_group.go:22:18 aws_db_instance_invalid_parameter_group.go:22:18 aws_elb_invalid_instance.go:23:18 aws_instance_previous_type.go:22:18 aws_alb_duplicate_name.go:22:18 aws_instance_invalid_key_name.go:22:18 aws_elasticache_cluster_invalid_subnet_group.go:22:18 aws_elb_invalid_security_group.go:23:18 aws_security_group_duplicate_name.go:24:18 aws_elb_invalid_subnet.go:23:18 aws_route_invalid_instance.go:22:18 aws_instance_invalid_ami.go:22:18 aws_db_instance_duplicate_identifier.go:22:18 detector.go:159:7 detector.go:234:7 aws_elasticache_cluster_default_parameter_group.go:19:18 aws_instance_invalid_vpc_security_group.go:23:18 aws_route_invalid_egress_only_gateway.go:22:18 aws_elasticache_cluster_invalid_type.go:22:18 aws_elasticache_cluster_previous_type.go:22:18 aws_instance_invalid_iam_profile.go:22:18 aws_db_instance_invalid_db_subnet_group.go:22:18 aws_route_invalid_route_table.go:22:18 aws_instance_default_standard_volume.go:16:18 aws_elasticache_cluster_invalid_security_group.go:23:18 aws_route_invalid_gateway.go:22:18 aws_route_invalid_vpc_peering_connection.go:22:18

<!-- Comment by [SideCI](https://sideci.com) -->